### PR TITLE
Make Region immutable

### PR DESCRIPTION
**Pull request by rmilfont**
Changes Region class to be immutable.

Commit:Mod

TaskNumber:#60

### CODE REVIEW

File: src/rbr/Region.java [hunk 1]
```diff
@@ -3,20 +3,20 @@
...
+    private final Set<Character> outputPorts;
+    private final Range box;
+    private final Set<Vertex> destinations;
 
     public Region(Set<Character> ip, Set<Vertex> destinations, Set<Character> op) {
         this.destinations = new HashSet<>(destinations);
         this.inputPorts = new HashSet<>(ip);
         this.outputPorts = new HashSet<>(op);
-        this.updateBox();
+        this.box = createBox();
```
Comment (jmferreiratech) [inline src/rbr/Region.java]: This method return something. Do not ask it to do something, instead ask for something. You could pass the destinations to make more sense.

File: src/rbr/Region.java [hunk 2]
```diff
@@ -3,20 +3,20 @@
...
     }
 
-    private void updateBox() {
+    private Range createBox() {
```
Comment (jmferreiratech) [inline src/rbr/Region.java]: It seems that we have some duplicated code:
https://github.com/LaNoC-UFC/rsbr/blob/develop/src/rbr/RBR.java#L182

**Review by jmferreiratech — CHANGES_REQUESTED** [review]

File: src/rbr/Region.java (hunk 1 above)
Comment (rmilfont) [reply src/rbr/Region.java]: Done.

File: src/rbr/Region.java (hunk 2 above)
Comment (rmilfont) [reply src/rbr/Region.java]: Put this method at EdgeColor class. After this, it was renamed to TopologyKnowledge, the idea is that class focus on the knowledge of the topology.

**Review by rafaelgmota — APPROVED** [review]

**Review by jmferreiratech — APPROVED** [review]

**Pull request merged**